atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/transaction/ReadTransactionAction.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.transaction;
+
+import com.palantir.atlasdb.workload.WorkloadCell;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+@Value.Immutable(builder = false)
+public interface ReadTransactionAction extends TransactionAction {
```

Comment:
For the time being, it's ok we only expect the `key` for our reads. Even though we're using dynamic columns, for time being it will be simply be a single put per key. Long-term I expect us to either introduce a  RangeColumnReadTransactionAction or something of the sort, or re-purpose this and expect _all_ reads to provide a column locator.